### PR TITLE
feat: Add data usage permissions (WEBAPP-5066 - WEBAPP-5071)

### PR DESCRIPTION
**Pull request by lipis**

### CODE REVIEW

File: app/script/view_model/content/PreferencesOptionsViewModel.js [hunk 1]
```diff
@@ -103,7 +98,7 @@ z.viewModel.content.PreferencesOptionsViewModel = class PreferencesOptionsViewMo
     this.optionAudio(properties.settings.sound.alerts);
     this.optionReplaceInlineEmoji(properties.settings.emoji.replace_inline);
     this.optionSendPreviews(properties.settings.previews.send);
-    this.optionPrivacy(properties.settings.privacy.improve_wire);
+    // this.optionPrivacy(properties.settings.privacy.improve_wire);
```
Comment (gregor) [inline app/script/view_model/content/PreferencesOptionsViewModel.js]: Remove excluded code

File: app/script/main/app.js [hunk 2]
```diff
@@ -310,6 +310,7 @@ z.main.App = class App {
         );
 
         this.repository.event_tracker.init(this.repository.properties.properties.settings.privacy.improve_wire);
+        this.repository.event_tracker.init(this.repository.properties.properties.settings.privacy.receive_news);
```
Comment (lipis) [inline app/script/main/app.js]: not sure if that should go here or not yet..

File: app/script/view_model/content/PreferencesAccountViewModel.js [hunk 3]
```diff
@@ -370,4 +386,9 @@ z.viewModel.content.PreferencesAccountViewModel = class PreferencesAccountViewMo
     this.enteredUsername(null);
     this.submittedUsername(null);
   }
+
+  updateProperties(properties) {
+    this.optionPrivacy(properties.settings.privacy.improve_wire);
+    this.optionReceiveNews(properties.settings.privacy.recieve_news);
```
Comment (gregor) [inline app/script/view_model/content/PreferencesAccountViewModel.js]: Typo: `receive_news`

File: app/page/template/content/preferences-account.htm [hunk 4]
```diff
@@ -83,6 +83,31 @@
...
+          <div class="preferences-option-icon checkbox text-theme" data-bind="attr: {'data-uie-value': optionReceiveNews}"  data-uie-name="status-preference-receive-news">
+            <input type="checkbox" id="receive-checkbox" data-bind="checked: optionReceiveNews">
+            <label class="preferences-options-checkbox-label" for="receive-checkbox" data-bind="l10n_text: z.string.preferencesAccountReceiveNewsCheckbox"></label>
+          </div>
+        </div>
+        <div class="preferences-detail" data-bind="l10n_text: z.string.preferencesAccountReceiveNewsDetail"></div>
+        <div class="preferences-link text-theme" data-bind="click: clickOnAccountExportInfo, l10n_text: z.string.preferencesAccountExportPersonalInfoHeadline" data-uie-name="do-backup-export"></div>
+        <div class="preferences-detail" data-bind="l10n_text: z.string.preferencesAccountExportPersonalInfoSecondary"></div>
+        <hr class="preferences-separator">
+        <div class="preferences-link text-theme" data-bind="click: clickOnReceiveNews">Show Receive News Confirm</div>
```
Comment (lipis) [inline app/page/template/content/preferences-account.htm]: Just a place holder for now to test the confirm form

File: app/script/main/app.js (hunk 2 above)
Comment (gregor) [reply app/script/main/app.js]: It should not. The event tracker should only be initialized once - with the privacy setting option. Not the GDPR email consent.

File: app/script/properties/PropertiesEntity.js [hunk 5]
```diff
@@ -35,6 +35,7 @@ z.properties.PropertiesEntity = class PropertiesEntity {
       },
       privacy: {
         improve_wire: true,
+        receive_news: true,
```
Comment (gregor) [inline app/script/properties/PropertiesEntity.js]: We definitely have to change these defaults.

Comment (lipis) [reply app/script/properties/PropertiesEntity.js]: yep.. it's just a placeholder.. open for suggestions :)

Comment (bennycode) [reply app/script/properties/PropertiesEntity.js]: Default will be `false` for `improve_wire` as discussed with @red-avtovo. The `receive_news` property needs to be removed from `PropertiesEntity.js` as we will get a dedicated endpoint for that.

File: app/script/auth/module/action/SelfAction.js [hunk 6]
```diff
@@ -53,3 +54,34 @@ export function setHandle(handle) {
...
+        throw BackendError.handle(error);
+      });
+  };
+}
+
+export function doSetConsent(consentType, value) {
+  return function(dispatch, getState, {apiClient}) {
+    dispatch(SelfActionCreator.startSetConsent());
+    const consent = {
+      source: `${APP_NAME}_${VERSION}`,
```
Comment (gregor) [inline app/script/auth/module/action/SelfAction.js]: Let's remove the underscore here.

**Pull request merged**